File: Reveal Cards In Increasing Order.cpp
class Solution {
public:
    vector<int> deckRevealedIncreasing(vector<int>& deck) {
     int n=deck.size();
     vector <int> ans (n);
     queue<int> q;
     sort(deck.begin(),deck.end());
     for (int i=0; i<deck.size();i++)
     {
            q.push(i);
     }  
     for (int i=0;i<n;i++){
         ans[q.front()]=deck[i];
         q.pop();
         if(!q.empty()){
             q.push(q.front());
             q.pop();
         }
     }
     return ans;
    }
};

File: pre-camp/week/leetcode/reveal-cards-in-increasing-order.py
class Solution:
    def deckRevealedIncreasing(self, deck: List[int]) -> List[int]:
        cards = deque()
        deck.sort(reverse = True)
        for i in range(len(deck)):
            if len(cards) == 0:
                cards.append(deck[i])
            else:
                x = cards.popleft()
                cards.append(x)
                cards.append(deck[i])
        return list(cards)[::-1]
            